packages/core/src/rum/instrumentation/resourceTracking/DdRumResourceTracking.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { InternalLog } from '../../../InternalLog';
+import { SdkVerbosity } from '../../../SdkVerbosity';
+
+import type { RequestProxy } from './domain/interfaces/RequestProxy';
+import { XHRProxy } from './implementation/XHRProxy';
+import { firstPartyHostsRegexBuilder } from './implementation/firstPartyHostsRegex';
+
+/**
+ * Provides RUM auto-instrumentation feature to track resources (fetch, XHR, axios) as RUM events.
+ */
+export class DdRumResourceTracking {
+    private static isTracking = false;
+    private static requestProxy: RequestProxy;
+
+    /**
+     * Starts tracking resources and sends a RUM Resource event every time a network request is detected.
+     */
+    static startTracking({
```

Comment:
Next step: remove this function that is there only for testing purposes, now we can do our tests on `XHRProxy` instead.